exporter/datadogexporter/metrics_exporter.go

```diff
@@ -26,23 +31,47 @@ type metricsExporter struct {
 	logger *zap.Logger
 	cfg    *Config
 	client *datadog.Client
-	tags   []string
 }
 
 func newMetricsExporter(logger *zap.Logger, cfg *Config) (*metricsExporter, error) {
 	client := datadog.NewClient(cfg.API.Key, "")
 	client.SetBaseUrl(cfg.Metrics.TCPAddr.Endpoint)
 
-	// Calculate tags at startup
-	tags := cfg.TagsConfig.GetTags(false)
+	return &metricsExporter{logger, cfg, client}, nil
+}
+
+// pushHostMetadata sends a host metadata payload to the "/intake" endpoint
+func (exp *metricsExporter) pushHostMetadata(metadata hostMetadata) error {
+	path := exp.cfg.Metrics.TCPAddr.Endpoint + "/intake"
+	buf, _ := json.Marshal(metadata)
+	req, _ := http.NewRequest(http.MethodPost, path, bytes.NewBuffer(buf))
+	req.Header.Set("DD-API-KEY", exp.cfg.API.Key)
+	req.Header.Set("Content-Type", "application/json")
+
+	client := &http.Client{Timeout: 10 * time.Second}
```

Comment:
Not a blocker for this PR, but we should investigate if we also need here all the options that the Agent passes as `Transport` on its http client. Maybe @olivielpeau can give his opinion on this. Links to the relevant parts in the Agent code:

`newHTTPClient()` which calls `CreateHTTPTransport()`:
https://github.com/DataDog/datadog-agent/blob/012a209d3acecf96fa74236cc5fc5a6d028a4daf/pkg/forwarder/worker.go#L55-L63

`CreateHTTPTransport()`:
https://github.com/DataDog/datadog-agent/blob/012a209d3acecf96fa74236cc5fc5a6d028a4daf/pkg/util/http/transport.go#L21-L55

This git blame might help us track why they were added in the Agent: https://github.com/DataDog/datadog-agent/blame/7305c07eaf917f63274749dfdd96119b0b713883/pkg/util/common.go

Eg: the PR that disables FallbackDelay mentions "Some concerns have been raised about the effect this might have on the intake."

Also, the proxy settings are probably something we also want.

Reply:
> we should investigate if we also need here all the options that the Agent passes as Transport on its http client

the zero value for the `http.Client` uses `http.DefaultTransport` that differs from the `Transport` the Agent uses in:

1. the `FallBackDelay` being disabled,
2. the `MaxIdleConnsPerHost` set to 5 in the Agent instead of the default of 2,
3. being able to disable TLS and
4. the Proxy settings being taken by default from the environment (which is what every other exporter does).

I think we should keep (4) as is and maybe change (1)/(2)/(3) depending on what Agent Core says. If we change (1)/(2)/(3) we should also change it on the metrics exporter client which uses [the default client](https://github.com/zorkian/go-datadog-api/blob/1da8da5725ee2cd0ac164baf69c8ec81cb4941df/client.go#L64) (maybe on zorkian's library?).

Reply:
Being consistent with how the proxy is configured everywhere else makes sense, I think `MaxIdleConnsPerHost` shouldn't matter and disabling TLS is not something we need to allow. The only thing is `FallBackDelay`, which I don't understand why was disabled in the Agent... but it can't be that bad if it's the default in Go :)

Reply:
1. we want to disable Fast Fallback as it can create unnecessary stress on Datadog's intake.
2. `MaxIdleConnsPerHost` set to `5` or `2` isn't that important, unless we expect a lot of concurrent connections to Datadog's intake from this client (not the case here), we want to use Go's default here, I think it changed across Go versions and we haven't updated it yet in `datadog-agent`.
3. we don't really allow disabling TLS on `datadog-agent`, but there's an opt-in to disable certificate validation, which can be useful for some specific proxy settings. There's also an option to force TLS 1.2 and up on the client side, which is a client-side enforcement some users require. Such settings may not be required now, but may be asked in the future.
4. agreed, I'm not aware of the specifics of how exporters should support proxy settings, but consistency with other exporters/the OT collector makes sense. Note that if you don't use the default transport, this means explicitly setting the `Proxy` field to `ProxyFromEnvironments`.

Also, `HTTP/2` is not enabled on most intake endpoints at the moment, so you can leave `ForceAttemptHTTP2` to `false`.